scanpipe/pipes/d2d_config.py

```diff
@@ -0,0 +1,185 @@
+    matchable_package_extensions: list = field(default_factory=list)
+
+    # These are extensions for file of this ecosystem that are
+    # matchable against the purldb using matchcode
+    matchable_resource_extensions: list = field(default_factory=list)
+
+    # Extensions for document files which do not require review
+    doc_extensions: list = field(default_factory=list)
+
+    # Paths in the deployed binaries/archives (on the to/ side) which
+    # do not need review even if they are not matched to the source side
+    deployed_resource_path_exclusions: list = field(default_factory=list)
+
+    # Paths in the development/source archive (on the from/ side) which
+    # should not be considered even if unmapped to the deployed side when
+    # assessing what to review on the deployed side
+    devel_resource_path_exclusions: list = field(default_factory=list)
+
+    # Symbols which are found in ecosystem-specific standard libraries
+    # which are not so useful in mapping
+    standard_symbols_to_exclude: list = field(default_factory=list)
+
+    # File extesions which should be looked at for source symbol extraction
+    # for mapping using symbols for a specific selected option/ecosystem
+    source_symbol_extensions: list = field(default_factory=list)
+
+
+# Dictionary of ecosystem configurations
+ECOSYSTEM_CONFIGS = {
+    "Default": EcosystemConfig(
+        matchable_package_extensions=[".zip", ".tar.gz", ".tar.xz"],
+        devel_resource_path_exclusions=["*/tests/*"],
+        doc_extensions=[
+            ".pdf",
+            ".doc",
+            ".docx",
+            ".ppt",
+            ".pptx",
+            ".tex",
+            ".odt",
+            ".odp",
+        ],
+    ),
+    "Java": EcosystemConfig(
+        ecosystem_option="Java",
+        matchable_package_extensions=[".jar", ".war"],
+        matchable_resource_extensions=[".class"],
+    ),
+    "JavaScript": EcosystemConfig(
+        ecosystem_option="JavaScript",
+        matchable_resource_extensions=[
+            ".map",
+            ".js",
+            ".mjs",
+            ".ts",
+            ".d.ts",
+            ".jsx",
+            ".tsx",
+            ".css",
+            ".scss",
+            ".less",
+            ".sass",
+            ".soy",
+        ],
+        source_symbol_extensions=[".ts", ".js"],
+    ),
+    "Go": EcosystemConfig(
+        ecosystem_option="Go",
+        matchable_resource_extensions=[".go"],
+    ),
+    "Rust": EcosystemConfig(
+        ecosystem_option="Rust",
+        matchable_resource_extensions=[".rs"],
+        source_symbol_extensions=[".rs"],
+    ),
+    "Ruby": EcosystemConfig(
+        ecosystem_option="Ruby",
+        matchable_package_extensions=[".gem"],
+        matchable_resource_extensions=[".rb"],
+        deployed_resource_path_exclusions=["*checksums.yaml.gz*", "*metadata.gz*"],
+    ),
+    "Elf": EcosystemConfig(
+        ecosystem_option="Elf",
+        source_symbol_extensions=[".c", ".cpp", ".h"],
```

Comment:
Please file an issue to improve this list which is way too small. Check the common or scancode function to see a list of these.

Reply:
Issue opened at https://github.com/aboutcode-org/scancode.io/issues/1707